src/test/scala/org.openeo/OpenSearchClientTest.scala

```diff
@@ -393,4 +393,24 @@ class OpenSearchClientTest {
     // This is the product we are actually interested in.
     features.find(_.generalProperties.processingBaseline.get == processingBaseline).get
   }
+
+  @Test
+  def parseCreodiasCorruptPhoebus(): Unit = {
+    HttpCache.enabled = true
+    val url = "https://finder.creodias.eu/oldresto/api/collections/Sentinel2/search.json?box=21.657597756412194%2C46.02854700799339%2C21.95285234099209%2C46.23461502351761&sortParam=startDate&sortOrder=ascending&page=1&maxRecords=100&status=0%7C34%7C37&dataset=ESA-DATASET&productType=L2A&cloudCover=%5B0%2C95%5D&startDate=2018-08-20T00%3A00%3A00Z&completionDate=2018-08-20T23%3A59%3A59.999999999Z"
+    val source = Source.fromURL(new URL(url))
+    val collectionsResponse = source.getLines.mkString("\n")
+    source.close()
```

Comment:
It could be better to use the 'using' construct instead of manually calling source.close()
```
  import scala.util.Using
  using(Source.fromURL(new URL(url))) { source =>
    val collectionsResponse = source.getLines.mkString("\n")
    val features = CreoFeatureCollection.parse(collectionsResponse, dedup = true).features
    
    for {
      f <- features
      l <- f.links
      if l.href.toString.contains("/PHOEBUS-core/")
    } {
      // There used to be a product with processingBaseline == 2.08
      throw new Exception("There should be no PHOEBUS-core in results.")
    }
    assertEquals(1, features.length)
  }
```